Composer/packages/lib/indexers/src/dialogIndexer.ts

```diff
@@ -38,14 +38,22 @@ function ExtractLgTemplates(dialog): string[] {
           return true;
       }
       targets.forEach(target => {
-        templates.push(...extractLgTemplateRefs(target).map(x => x.name));
+        // match a template name match a temlate func  e.g. `showDate()`
+        // eslint-disable-next-line security/detect-unsafe-regex
+        const reg = /\[([A-Za-z_][-\w]+)(\(.*\))?\]/g;
+        let matchResult;
+        while ((matchResult = reg.exec(target)) !== null) {
+          const templateName = matchResult[1];
+          templates.push(templateName);
+        }
```

Comment:
Why don't we want to use `extractLgTemplateRefs`?

Reply:
I have synced with @yeze322 @zhixzhan. I can implement string => name function separately in indexer.
We want to make indexer independent. If use `extractLgTemplateRefs`, indexer will depend on shared.

@boydc2014 Do you have any comments about this.

Reply:
My suggestion is move some utils in shared into indexers if don't want indexers depends on shared, and re-export in shared for any upper layer using shared. 

To clarify, the reason we don't want indexers depends on shared, we want the vice-versa, is that indexers (including the validation inside) can be used by command line tool like "DialogLint". It's would be good we keep the pure "Declartive" world's concept in indexers. and let "shared" package be a shared package in Composer's world.

Reply:
Given "extractLgTemplateRefs" is only used here, you can just move it here without re-export, then you need to move "LgTemplatRef" interface here and re-export.

Reply:
I have moved all lgutils to indexers.